sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -97,6 +102,24 @@
             .typeSystem(connection.getTypeFactory().getTypeSystem())
             .operatorTable(ChainedSqlOperatorTable.of(opTab0, catalogReader))
             .build();
+
+    plannerImplConstructor =
+        getPlannerConstructorFromPipelineOptions(PipelineOptionsFactory.create());
+  }
+
+  public static Constructor getPlannerConstructorFromPipelineOptions(PipelineOptions options) {
+    String plannerImplClassName = null;
+    try {
+      plannerImplClassName = options.as(BeamSqlPipelineOptions.class).getPlannerImplClassName();
+      return Class.forName(plannerImplClassName).getConstructor(FrameworkConfig.class);
+    } catch (NoSuchMethodException | ClassNotFoundException e) {
```

Comment:
These need to be separate messages:
 * Couldn't the planner implementation class '...' specified in the options;
 * Couldn't find a constructor that accepts `FrameworkConfig` parameter;